blocks/basic/include/gnuradio-4.0/basic/DataSink.hpp

```diff
@@ -181,45 +207,45 @@ class DataSinkRegistry {
     }
 
     template<typename T>
-    std::shared_ptr<StreamingPoller<T>> getStreamingPoller(const DataSinkQuery& query, BlockingMode block = BlockingMode::Blocking) {
+    std::shared_ptr<StreamingPoller<T>> getStreamingPoller(const DataSinkQuery& query, std::size_t minRequiredSamples = 1UZ, std::size_t maxRequiredSamples = std::numeric_limits<std::size_t>::max(), BlockingMode blockingMode = BlockingMode::Blocking) {
```

Comment:
Proposal to change this to a config object that allows also further extensions, i.e.

```cpp
struct PollerConfig {
    BlockingMode blockingMode       = BlockingMode::Blocking; /// rename enum-values to: 'ApplyBackpressue' & `DropSamples`
    std::size_t  minRequiredSamples = 1UZ;   /// brief explaination (i.e. optimises throughput)
    std::size_t  maxRequiredSamples = std::numeric_limits<std::size_t>::max();  /// brief explanation (i.e. optimises latency)
    // other pertinent parameters for the other pollers.
};

std::shared_ptr<StreamingPoller<T>> getStreamingPoller(const DataSinkQuery& query, PollerConfig config = {}) { ..}
```